cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_hsw.cpp

```diff
@@ -31,17 +30,308 @@ namespace oneapi::dal::preview {
 namespace jaccard {
 namespace detail {
 
-template vertex_similarity_result
-call_jaccard_default_kernel_avx2<oneapi::dal::backend::cpu_dispatch_avx2>(
-    const descriptor_base &desc,
-    vertex_similarity_input<undirected_adjacency_array_graph<>> &input);
+#if defined(__INTEL_COMPILER)
+DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
+    return _popcnt32(x);
+}
+#else
+DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
+    std::int32_t count = 0;
+    std::int32_t a = x;
+    while (a != 0) {
+        a = a & (a - 1);
+        count++;
+    }
+    return count;
+}
+#endif
```

Comment:
```suggestion
#if defined(__INTEL_COMPILER)
DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
    return _popcnt32(x);
}
#elif defined(_MSC_VER)
DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
    return __popcnt(x);
}
#elif (defined(__GNUC__) || defined(__clang__)) && ((defined(__has_builtin) && __has_builtin(__builtin_popcnt)) || defined(__builtin_popcnt))
DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
    return __builtin_popcount(x);
}
#else
DAAL_FORCEINLINE std::int32_t _popcnt32_redef(const std::int32_t &x) {
    std::int32_t count = 0;
    std::int32_t a = x;
    while (a != 0) {
        a = a & (a - 1);
        count++;
    }
    return count;
}
#endif
```
Docs:
1. https://docs.microsoft.com/en-us/cpp/intrinsics/popcnt16-popcnt-popcnt64?view=vs-2019
2. https://clang.llvm.org/docs/LanguageExtensions.html
3. https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html
4. https://gcc.gnu.org/onlinedocs/gcc-10.1.0/cpp/_005f_005fhas_005fbuiltin.html

Actually situation with `__has_builtin` is a little bit complicated since it's deprecated in Clang since 10.0 version (they recommend to use `defined(__builtin_NAME)` instaed) and added into GCC spec since 10 version (magic of number 10).